conceptual/Npgsql/release-notes/5.0.md

```diff
@@ -2,11 +2,34 @@
+* *Connection multiplexing* ([#1982](https://github.com/npgsql/npgsql/issues/1982)). Multiplexing is an experimental new way for the driver to handle commands and connections, leading to a potentially significant reduction in PostgreSQL physical connection requirements. This feature is also providing a substantial performance boot on the [TechEmpower benchmark](https://www.techempower.com/benchmarks/) (next round to be published). Multiplexing is disabled by default, and must be opted into.; full documentation will be available soon.
+* All Npgsql APIs are now fully annotated for *C# nullable references* ([#3120](https://github.com/npgsql/npgsql/issues/3120)), in line with the new annotations in System.Data released in .NET 5.0.
+* Reading PostgreSQL arrays as CLR *arrays of nullable value types* (e.g. `int?[]` is now supported ([#443](https://github.com/npgsql/npgsql/issues/443)). Previously, only non-nullable value arrays were supported, and nulls were translated to the default value (e.g. 0).
+* The connection pool can be configured to unconditionally *close connections which have reached a certain age* ([#1810](https://github.com/npgsql/npgsql/issues/1810)). This can useful to rebalance connections in a distributed database scenario, or to mitigate certain resource leaks.
 
 ## Breaking changes
 
-* The *TcpKeepAliveTime* and *TcpKeepAliveInterval* connection string parameters are now expressed in seconds instead of milliseconds.
+* Npgsql no longer targets .NET Framework 4.6.1. Since .NET Standard 2.0 is targeted, it is still possible to use Npgsql from .NET Framework applications; however, we no longer run regression tests on .NET Framework and will only fix bugs on a best-effort basis. In addition, the Visual Studio extension (VSIX) and the MSI GAC installer have been discontinued. [#3269](https://github.com/npgsql/npgsql/issues/3269).
+* Npgsql targets the `netcoreapp3.1` TFM instead of `netcoreapp3.0`, which is out of support ([#3160](https://github.com/npgsql/npgsql/issues/3160)).
+* When command timeout occurs, Npgsql now raises an <xref:Npgsql.NpgsqlException> wrapping a [TimeoutException](https://docs.microsoft.com/dotnet/api/system.timeoutexception) ([#3132](https://github.com/npgsql/npgsql/issues/3132)). Previously, an NpgsqlException wrapping an IOException was raised.
```

Comment:
Shouldn't we add the same for the cancellation - we now throw `OperationCancelledException`, instead of `PostgresException`.

Reply:
Yep, thanks.

Reply:
I plan to do a separate page with more thorough documentation on timeouts, cancellations and keepalive - we should link to there for more details.